osu.Game/Screens/Select/SongSelect.cs

```diff
@@ -307,6 +307,19 @@ protected override void OnEntering(Screen last)
             Content.FadeInFromZero(250);
 
             FilterControl.Activate();
+
+            // select and preview a random beatmap if we come from MainMenu the first time
+            if (last is MainMenu)

```

Comment:
I'm pretty sure you can just compare `Current` to `BeatmapManager.DefaultBeatmap` instead of exposing the hash and stuff (just expose a `bool` in `BeatmapManager` `IsCurrentDefault`)

Reply:
Also this should work fine for all display modes using the random function in SongSelect (and would be more correct because it'll only select form visible beatmaps).